clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -18197,7 +18197,7 @@ Value *CodeGenFunction::EmitPPCBuiltinExpr(unsigned BuiltinID,
       CallOps.push_back(Ops[i]);
     llvm::Function *F = CGM.getIntrinsic(ID);
     Value *Call = Builder.CreateCall(F, CallOps);
-    return Builder.CreateAlignedStore(Call, Ops[0], MaybeAlign(64));
+    return Builder.CreateAlignedStore(Call, Ops[0], MaybeAlign());
```

Comment:
```suggestion
    return Builder.CreateStore(Call, Ops[0]);
```
Should work, I think? No need to use CreateAlignedStore if you want a naturally aligned store.

Reply:
I'd prefer to just explicitly write the expected alignment here, though; LLVM DataLayout rules for alignment can be a bit strange.

Reply:
> I'd prefer to just explicitly write the expected alignment here,

@efriedma-quic  Do you mean like hard code the expected alignment?

Currently `CreateAlignedStore(Call, Ops[0], MaybeAlign());` calls:
```
StoreInst *CreateAlignedStore(Value *Val, Value *Ptr, MaybeAlign Align,
                                bool isVolatile = false) {
    if (!Align) {
      const DataLayout &DL = BB->getDataLayout();
      Align = DL.getABITypeAlign(Val->getType());
    }
    return Insert(new StoreInst(Val, Ptr, isVolatile, *Align));
  }
```
and get the proper ABI alignment for the type given.  Since this section of code process types that are either 32bit or 64bit aligned, we can't hard code any alignment here.

Reply:
Oh, didn't realize there were multiple different types coming through here.  It's not worth it to duplicate the implementation of CreateStore... @nikic suggestion is fine, then.

Reply:
@nikic Using `CreateStore(Value, Address)` will require extra code to convert`Ops[0[` to type `Address`.

Reply:
Ah, it looks like CGBuilder doesn't directly re-export this IRBuilder API, it's called CreateDefaultAlignedStore() instead.

Reply:
I see a FIXME for this function, do we still want to use it?
```
  // FIXME: these "default-aligned" APIs should be removed,
  // but I don't feel like fixing all the builtin code right now.
  llvm::StoreInst *CreateDefaultAlignedStore(llvm::Value *Val,
                                             llvm::Value *Addr,
                                             bool IsVolatile = false) 
```
Since these need to be aligned stores, maybe it's more clear to just use `CreateAlignedStore()`?
The patch that added this FIXME:
```
commit 7f416cc426384ad1f891addb61d93e7ca1ffa0f2
Author: John McCall <rjmccall@apple.com>
Date:   Tue Sep 8 08:05:57 2015 +0000
```
Explicitly asks:
```
    I partially punted on applying this work to CGBuiltin.  Please
    do not add more uses of the CreateDefaultAligned{Load,Store}
    APIs; they will be going away eventually.

    llvm-svn: 246985
```
It is an old patch so I am not sure how much of that is still true...

Reply:
Oh, right, I forgot we were trying to get rid of those at one point... that's probably never happening for LLVM in general, but maybe for clang it'll get finished at some point.

Looking at the code again, it's calling EmitPointerWithAlignment, so you can just take the alignment from that.  Well, actually, it looks like there's a bug where we emit the first argument twice, but with that fixed, you should be able to do that.